File: basemodule/src/main/java/com/appoets/basemodule/base/BaseBottomSheet.java
package com.appoets.basemodule.base;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.appoets.basemodule.R;
import com.google.android.material.bottomsheet.BottomSheetDialogFragment;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.databinding.DataBindingUtil;
import androidx.databinding.ViewDataBinding;
import androidx.lifecycle.ViewModel;

public abstract class BaseBottomSheet<T extends ViewDataBinding> extends BottomSheetDialogFragment {

    protected  abstract  void initView(View View);
    private  View rootView;
    private  T mViewDataBinding;
    private  View view;

    public  abstract  int getLayout();
    public  abstract  void initView(ViewDataBinding mViewDataBinding);

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        mViewDataBinding = DataBindingUtil.inflate(inflater,getLayout(), container, false);
        initView(mViewDataBinding);
        view=mViewDataBinding.getRoot();
        return view;
    }
}

File: base/src/main/java/com/gox/base/base/BaseBottomSheet.kt
package com.gox.base.base

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.annotation.LayoutRes

import com.google.android.material.bottomsheet.BottomSheetDialogFragment
import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding

abstract class BaseBottomSheet<T : ViewDataBinding> : BottomSheetDialogFragment() {

    private var mViewDataBinding: T? = null

    @LayoutRes
    protected abstract fun getLayoutId(): Int

    abstract fun initView(mViewDataBinding: ViewDataBinding?)

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        mViewDataBinding = DataBindingUtil.inflate(inflater, getLayoutId(), container, false)
        initView(mViewDataBinding)
        return view
    }
}
